jmsSource/src/main/java/io/vantiq/extsrc/jmsSource/communication/JMSMessageListener.java

```diff
@@ -0,0 +1,136 @@
+        this.context = context;
+        this.client = client;
+        this.messageHandler = messageHandler;
+    }
+    
+    /**
+     * A method used to setup the MessageListener for the given destination (topic or queue)
+     * @param connectionFactoryName     The name of the connection factory used to connect to the JMS Server
+     * @param dest                      The name of the destination to connect to (topic or queue)
+     * @param isQueue                   A boolean flag used to create the appropriate type of destination (queue or topic)
+     * @param username                  The username used to create the JMS Connection, (or null if JMS Server does not require auth)
+     * @param password                  The password used to create the JMS Connection, (or null if JMS Server does not require auth)
+     * @throws NamingException
+     * @throws JMSException
+     * @throws FailedJMSSetupException
+     */
+    public synchronized void open(String connectionFactoryName, String dest, boolean isQueue, String username, String password) throws NamingException, JMSException, FailedJMSSetupException {
+        this.destName = dest;
+        
+        connectionFactory = (ConnectionFactory) context.lookup(connectionFactoryName);
+        if (connectionFactory == null) {
+            throw new FailedJMSSetupException("The Connection Factory named " + connectionFactoryName + " was unable to be found.");
+        }
+        
+        connection = connectionFactory.createConnection(username, password);
+        if (connection == null) {
+            throw new FailedJMSSetupException("A Connection was unable to be created using the Connection Factory named " + connectionFactoryName + ".");
+        }
+        
+        session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+        if (session == null) {
+            throw new FailedJMSSetupException("A Session was unable to be created.");
+        }
+        
+        if (isQueue) {
+            destination = session.createQueue(dest);
+        } else {
+            destination = session.createTopic(dest);
+        }
+        if (destination == null) {
+            throw new FailedJMSSetupException("A Destination with name " + dest + " was unable to be created.");
+        }
+        
+        MessageConsumer consumer = session.createConsumer(destination);
+        if (consumer == null) {
+            throw new FailedJMSSetupException("A Message Producer for the Destination with name " + dest + " was unable to be created.");
+        }
+        
+        consumer.setMessageListener(this);
+        connection.start();
+    }
+    
+    /**
+     * The method used to handle incoming messages. Sends the formatted message to VANTIQ as a Source Notification.
+     */
+    @Override
+    public void onMessage(Message msg) {
+        try {
+            Map<String, Object> msgMap = messageHandler.parseIncomingMessage(msg, destName);
+            client.sendNotification(msgMap);
```

Comment:
Is there anything here in terms of required properties necessary in the returned `msgMap`?  the format or anything?  If so, we should check them here & document what's required.

From the code, it looks like we expect `message` (possibly optional), `JMSFormat`, and `destination`, and this matches the example code (not the text) in the readme.

Before, returning, we should verify that we go what we expected -- otherwise, complain.